### PR TITLE
chore: setup CircleCI

### PR DESCRIPTION
**Pull request by faustbrian**
@ItsANameToo you can work on the config on the `circle` branch

### CODE REVIEW

**Comment by faustbrian** [comment]
CircleCI is now enabled so it runs the tests for this branch.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      xcode: "10.1.0"
+    working_directory: ~/arkecosystem/client
+    steps:
+      - checkout
+      - run:
+          name: Install prerequisits
+          command: |
+            brew install carthage || true
+            brew outdated carthage || brew upgrade carthage
+            gem install xcpretty-travis-formatter --no-rdoc --no-ri --no-document --quiet
```
Comment (faustbrian) [inline .circleci/config.yml]: Is this still needed on CircleCI?

Comment (ItsANameToo) [reply .circleci/config.yml]: well you removed it now anyway

File: .circleci/config.yml [hunk 2]
```diff
@@ -11,15 +11,12 @@ jobs:
           command: |
             brew install carthage || true
             brew outdated carthage || brew upgrade carthage
-      - run:
-          name: Install SwiftLint
-          command: wget --output-document /tmp/SwiftLint.pkg https://github.com/realm/SwiftLint/releases/download/0.26.0/SwiftLint.pkg && sudo installer -pkg /tmp/SwiftLint.pkg -target /
```
Comment (ItsANameToo) [inline .circleci/config.yml]: We will still need swiftlint, just need to include it in another way since circleci works with jobs for that

**Comment by faustbrian** [comment]
```
*** Building scheme "Alamofire watchOS" in Alamofire.xcworkspace
*** Building scheme "Alamofire tvOS" in Alamofire.xcworkspace
```

Could we somehow limit what OS gets build for Alamofire? watchOS and tvOS seem pointless.

File: .circleci/config.yml (hunk 2 above)
Comment (faustbrian) [reply .circleci/config.yml]: Yeah, just removed it to see if it works otherwise. Should be added in a lint job.

**Comment by ItsANameToo** [comment]
yes, should be a `--platform` option

**Pull request merged**